internal/services/network/subnet_service_endpoint_storage_policy_resource.go

```diff
@@ -225,19 +242,22 @@ func flattenServiceEndpointPolicyDefinitions(input *[]network.ServiceEndpointPol
 
 		var (
 			description     = ""
+			service         = ""
 			serviceResource = []interface{}{}
 		)
 		if b := e.ServiceEndpointPolicyDefinitionPropertiesFormat; b != nil {
 			if b.Description != nil {
 				description = *b.Description
 			}
 			serviceResource = utils.FlattenStringSlice(b.ServiceResources)
+			service = *b.Service
```

Comment:
we'll need to nil check this to prevent a possible panic

Reply:
Thanks @katbyte , I've updated the PR to include the nil check. Ready for re-review.

Reply:
also tests are passing with the new change
```
2023-05-24T20:45:17.9407850Z TF_ACC=1 go test -v ./internal/services/network -run=TestAccSubnetServiceEndpointStoragePolicy -timeout 600m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
2023-05-24T20:54:06.5311822Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_basic
2023-05-24T20:54:06.5381341Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_basic
2023-05-24T20:54:06.5381709Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_complete
2023-05-24T20:54:06.5388551Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_complete
2023-05-24T20:54:06.5388837Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_alias
2023-05-24T20:54:06.5389080Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_alias
2023-05-24T20:54:06.5408502Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_storage
2023-05-24T20:54:06.5408713Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_storage
2023-05-24T20:54:06.5413236Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_update_complete
2023-05-24T20:54:06.5416648Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_update_complete
2023-05-24T20:54:06.5416911Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_update_alias
2023-05-24T20:54:06.5421280Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_update_alias
2023-05-24T20:54:06.5421535Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_update_storage
2023-05-24T20:54:06.5421737Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_update_storage
2023-05-24T20:54:06.5425504Z === RUN   TestAccSubnetServiceEndpointStoragePolicy_requiresImport
2023-05-24T20:54:06.5425781Z === PAUSE TestAccSubnetServiceEndpointStoragePolicy_requiresImport
2023-05-24T20:54:06.5425975Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_complete
2023-05-24T20:54:06.5428658Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_basic
2023-05-24T20:55:16.9782811Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_basic (70.55s)
2023-05-24T20:55:16.9783535Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_requiresImport
2023-05-24T20:55:44.4541566Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_complete (96.94s)
2023-05-24T20:55:44.4589388Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_update_storage
2023-05-24T20:56:10.8374144Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_requiresImport (53.86s)
2023-05-24T20:56:10.8374904Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_update_alias
2023-05-24T20:57:54.2944634Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_update_storage (130.93s)
2023-05-24T20:57:54.2945014Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_update_complete
2023-05-24T20:58:08.6247642Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_update_alias (117.79s)
2023-05-24T20:58:08.6290260Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_storage
2023-05-24T20:59:31.7616736Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_storage (83.14s)
2023-05-24T20:59:31.7617452Z === CONT  TestAccSubnetServiceEndpointStoragePolicy_alias
2023-05-24T21:00:06.6398108Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_update_complete (132.35s)
2023-05-24T21:00:39.3536489Z --- PASS: TestAccSubnetServiceEndpointStoragePolicy_alias (67.59s)
2023-05-24T21:00:39.3537168Z PASS
2023-05-24T21:00:39.3641724Z ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/network	392.953s
2023-05-24T21:00:40.8266243Z ##[section]Finishing: make acctests
```